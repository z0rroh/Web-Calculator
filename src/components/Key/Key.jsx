import React from 'react';
import PropTypes from 'prop-types';
import './Key.css'

const Key = ({ keyType, keyValue, keyAction}) => {

   return(
      <div 
         onClick={() => keyAction(keyValue)}
         className={`key-container ${keyType}`}>
         <p className="key-value">
            { keyValue }
         </p>
      </div>
   )
}

Key.propTypes = {
   keyAction: PropTypes.func.isRequired,
   keyType: PropTypes.string.isRequired,
   keyValue: PropTypes.string.isRequired
}

export default Key