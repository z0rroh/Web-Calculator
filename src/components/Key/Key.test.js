import React from 'react';
import { shallow } from 'enzyme'
import Key from './Key'

describe('<Key/>', () => {
   let wrapp

   beforeEach(() => {
      wrapp = shallow(
         <Key
            keyAction={jest.fn()}
            keyType={''}
            keyValue={''}
         />
      )
   });

   it('Should render correctly', () => {
      expect(wrapp).toMatchSnapshot()
   })

   it('Should render a div', () =>Â {
      expect(wrapp.find('div').length).toEqual(1)
   })

   it('should render the value of keyValue', () => {
      wrapp.setProps({ keyValue: 'test'})
      expect(wrapp.text()).toEqual('test')
   })
});
