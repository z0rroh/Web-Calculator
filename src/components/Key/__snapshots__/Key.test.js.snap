// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Key/> Should render correctly 1`] = `
<div
  className="key-container "
  onClick={[Function]}
>
  <p
    className="key-value"
  />
</div>
`;
